src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -572,6 +583,13 @@ void LinearIR::enumerate_expressions() const {
     }
 }
 
+void LinearIR::sort_results() {
```

Comment:
I'm not sure how that impacts performance, but is it possible to check if we have changed result node and only in this case perform the sort. What do you think?

Reply:
Result number is not large for most case, but yes, is_sorted check is better. Added, thanks!